libbeat/reader/syslog/message.go

```diff
@@ -102,13 +106,15 @@ func (m *message) setTimestampBSD(v string, loc *time.Location) {
 	}
 	if t, err := time.ParseInLocation(time.Stamp, v, loc); err == nil {
 		t = t.AddDate(time.Now().In(loc).Year(), 0, 0)
+		m.timestampSet = true
 		m.timestamp = t
 	}
 }
 
 // setPriority sets the priority for this message. The facility and severity are
 // derived from the priority and associated values are set.
 func (m *message) setPriority(v string) {
+	m.prioritySet = true
```

Comment:
If priority is preset to -1, that can be used as a natural flag for having been set by the parser.